tests/Api/Controller/ListGroupsControllerTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Tests\Api\Controller;
+
+use Flarum\Api\Controller\ListGroupsController;
+use Flarum\Group\Group;
+
+class ListGroupsControllerTest extends ApiControllerTestCase
+{
+    protected $controller = ListGroupsController::class;
+
+    /**
+     * @test
+     */
+    public function shows_index_for_guest()
+    {
+        $response = $this->callWith();
+
+        $this->assertEquals(200, $response->getStatusCode());
+        $data = json_decode($response->getBody()->getContents(), true);
+
+        $this->assertEquals(Group::count(), count($data['data']));
```

Comment:
Shouldn't this at least set up a group or something? I'd rather have the expected value be a fixed value, e.g. three groups.

Reply:
Yes we should go deeper into confirming actual values deeper down. Sorry I didn't have the time to patch this up yet before you merged.

Reply:
No worries. Want to improve the infrastructure first. :)